docs/testing/e2e-testing-standards.md

```diff
@@ -0,0 +1,262 @@
+
+        - `config/`: Configuration files for Playwright and other tools.
+
+            - `playwright.config.ts`: Playwright configuration.
+
+        - `hooks/`: Cucumber hooks for setting up and tearing down test environments.
+
+            - `hooks.ts`: Contains `Before`, `After`, and other lifecycle hooks.
+
+        - `reports/`: Generated test reports (e.g., HTML, JSON).
+
+        - `screenshots/`: Captured screenshots during test execution.
+
+        - `videos/`: Recorded videos of test runs.
+
+        - `test-data/`: Static test data files or files for upload.
+
+            - `data.json`: Example JSON files or CSVs with test data.
+
+
+
+## Test Structure - Arrange, Act, Assert:
+
+We can follow the AAA (Arrange, Act, Assert) pattern when structuring the tests. In most cases, the Arrange step can be included in a Before block(hook).
+Consider including comments defining each section to ease readability.
+
+```typescript
+// arrange, create a let propert
+await createProperty()
+
+// act, raise a charge
+// here we can call the method(s) defined in page objects too
+await raiseCharge()
+
+// assert, confirm charge has been raised
+expect(charge).toBe('raised')
+```
+
+## Page Object Model (POM)
+
+Every page should possess one POM file to enhance maintainability and scalability of our tests. It should include all the selectors and functions which belong to corresponding POM.
+
+All interactions should be done using the page objects, no selectors in your tests.
+
+All assertions should be in your test, no assertion in the POM
+
+DO 👍
+
+```typescript
+// POM file './pageOobjects/foo/'
+// add all locators and functions related to the page. 
+// allowing all tests to reuse 
+
+import {expect, Locator, Page} from '@playwright/test';
+
+export class FooPage {
+    readonly errorMessage: Locator;
+
+    constructor(page: Page) {
+        this.page = page;
+        this.errorMessage = page.locator('.error-message');
+    }
+}
+
+// test file './steps/foo.ts'
+import {FooPage} from './pageObjects/foo';
+
+let fooPage: FooPage;
+
+Then(
+    'error message should be visible',
+    async function ({page}) {
+        const fooPage = new FooPage({page});
+        await expect(fooPage.errorMessage).toBeVisible();
+    }
+)
+```
+
+DO NOT ⚔️
+```typescript
+// test file './steps/foo.ts'
+// include loctors directly in test
+import { Locator, Page } from '@playwright/test';
+
+Then(
+    'error message should be visible',
+    async function ({page}) {
+        await expect(page.locator('.error-message')).toBeVisible();
+    }
+)
+```
+
+## Waiting:
+
+We can avoid using any arbitrary waits until it is really, really necessary.
+
+DO NOT ⚔️
+
+```typescript
+await page.waitForTimeout(5000)
+```
+
+This can cause flaky tests as we can rarely be certain the amount of wait time is enough. It can also unnecessarily increase the test run time . Instead, we can try:
```

Comment:
```suggestion
This can cause flaky tests as we can rarely be certain the amount of wait time is enough. It can also unnecessarily increase the test run time. Instead, we can try:
```